packages/slate-react/src/hooks/android-input-manager/android-input-manager.ts

```diff
@@ -396,25 +396,32 @@ export function createAndroidInputManager({
     let canStoreDiff = true
 
     if (type.startsWith('delete')) {
-      if (Range.isExpanded(targetRange)) {
-        const [start, end] = Range.edges(targetRange)
-        const leaf = Node.leaf(editor, start.path)
+      const direction = type.endsWith('Backward') ? 'backward' : 'forward'
+      let [start, end] = Range.edges(targetRange)
+      let [leaf, path] = Editor.leaf(editor, start.path)
```

Comment:
Just moved these up instead of having separate definitions inside the branch and after it. We just make sure to update them if we make any changes